platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java

```diff
@@ -288,6 +288,28 @@ private void updateProperty(String key, Map<String, String> updatedProperties,
         }
     }
 
+    private void updatePortProperty(Map<String, String> updatedProperties,
```

Comment:
If you called this method after all the other system properties have been set you could simplify this method by retrieving the protocol and port numbers using the SystemBaseUrl helpers which will never return null.